### PR TITLE
feat(back-end): add status code on each route

### PR DESCRIPTION
**Pull request by AdrieneTZ**
- Add status code on each route (meet issue #10 )
- All routes have passed the thunder client test

### CODE REVIEW

**Review by davipon — CHANGES_REQUESTED** [review]
Please add a status code like `400` to handle the bad request.

**Comment by AdrieneTZ** [comment]
- Add a status code on each `res`, including checking conditions to handle the bad request.
- In the controller of `register`, if the email address has been registered, I'm not sure this `res` should end with status code `507` or `400`.

File: back-end/controllers/record-controller.js [hunk 1]
```diff
@@ -42,30 +44,32 @@ const recordController = {
...
+          .json('missing title or amount or category id to update this record')
 
       // Check if the record is in database
       const theRecord = await prisma.record.findUnique({
         where: {
           id: recordId,
         },
       })
-      theRecord ? theRecord : res.json('the record does not exist')
+      theRecord ? theRecord : res.status(400).json('the record does not exist')
```
Comment (davipon) [inline back-end/controllers/record-controller.js:57]: I assume that you want to check if `theRecord` exists (not `undefined`) or return status code `400`.

If that's the case, I'd suggest rewriting like this:

```js
if (!theRecord) res.status(400).json('the record does not exist')
```

It's clear that you're checking the result of the prisma query.

File: back-end/controllers/record-controller.js [hunk 2]
```diff
@@ -85,12 +89,14 @@ const recordController = {
...
```
Comment (davipon) [inline back-end/controllers/record-controller.js:92]: Same as line: 57.

File: back-end/controllers/record-controller.js [hunk 3]
```diff
@@ -42,30 +44,32 @@ const recordController = {
...
 
       // Check if the record is one of the current user's records
       userId === theRecord.userId
         ? userId
-        : res.json('this record does not belong to the current user')
+        : res
+            .status(400)
+            .json('this record does not belong to the current user')
```
Comment (davipon) [inline back-end/controllers/record-controller.js:64]: Same as line: 57.

File: back-end/controllers/record-controller.js [hunk 4]
```diff
@@ -85,12 +89,14 @@ const recordController = {
...
```
Comment (davipon) [inline back-end/controllers/record-controller.js:99]: Same as line: 57

File: back-end/controllers/user-controller.js [hunk 5]
```diff
@@ -19,15 +20,18 @@ const userController = {
           email: email,
         },
       })
+
       user
         ? user
-        : res.json('Incorrect email address or this user has not registered.')
+        : res
+            .status(400)
+            .json('Incorrect email address or this user has not registered.')
```
Comment (davipon) [inline back-end/controllers/user-controller.js:28]: Same as line 57 in `record-controller.js`.

**Review by davipon — CHANGES_REQUESTED** [review]
There is inconsistent style when handling `res.status()` or `res.json()` in controllers:

```js
// some controllers
return res.status(200)

// other controllers
res.status(200)
```

They might both work, but a consistent coding style can improve code quality over time.

File: back-end/controllers/record-controller.js [hunk 6]
```diff
@@ -39,33 +41,35 @@ const recordController = {
       const recordId = Number(req.params.rid)
       const { title, amount, note, categoryId } = req.body
       const userId = req.user.id
+      console.log(userId)
```
Comment (davipon) [inline back-end/controllers/record-controller.js:44]: Better remove this, I'd approve anyway. :)

**Review by davipon — APPROVED** [review]
LGTM

File: back-end/controllers/record-controller.js (hunk 6 above)
Comment (AdrieneTZ) [reply back-end/controllers/record-controller.js:44]: Oh, noooooo
I didn't notice this line.

**Pull request merged**